dmaws/rds.py

```diff
@@ -248,13 +248,24 @@ def clean_database(self):
         )
 
         open_framework_ids = self.get_open_framework_ids()
-        self.log("Delete draft services for open frameworks")
+
+        self.log("Delete draft services ")
+        self.log("  > Delete draft services for open frameworks")
         self.cursor.execute("""
             DELETE FROM draft_services WHERE framework_id IN (
                 SELECT id FROM frameworks WHERE status='open'
             )
             """)
 
+        self.log("  > Delete draft services belonging to dangling suppliers")
+        self.cursor.execute("""
+            DELETE FROM draft_services WHERE supplier_id IN (
```

Comment:
I think this could be written as 
```sql
DELETE FROM draft_services WHERE supplier_id NOT IN (
    SELECT supplier_id FROM suppliers INNER JOIN supplier_frameworks on suppliers.supplier_id=supplier_frameworks.supplier_id
)
```

Not sure if it's batter. Or whether this matters at all.